sys/include/riot_hdr.h

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2018 Kaspar Schleiser <kaspar@schleiser.de>
+ *                    Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_riot_hdr RIOT header helpers and tools
+ * @ingroup     sys
+ * @{
+ *
+ * The header contains
+ *
+ * - "RIOT" as magic number
+ * - the application version
+ * - the address where to find the RIOT firmware
+ * - the checksum of the three previous fields
+ *
+ * @file
+ * @brief       RIOT "partition" header and tools
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ * @author      Francisco Acosta <francisco.acosta@inria.fr>
+ *
+ * @}
+ */
+
+#ifndef RIOT_HDR_H
+#define RIOT_HDR_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdint.h>
+
+/**
+ * @brief  Magic number for riot_hdr
+ *
+ */
+#define RIOT_HDR_MAGIC         0x544f4952 /* "RIOT" */
+
+/**
+ * @brief Structure to store image header - All members are little endian
+ * @{
+ */
+typedef struct {
+    uint32_t magic_number;     /**< header magic_number (always "RIOT")              */
+    uint32_t version;          /**< Integer representing the partition version       */
+    uint32_t start_addr;       /**< Address after the allocated space for the header */
```

Comment:
The reason is that manipulating uint32_t variables is far easier than pointers (and it's arithmetic, since we do several operations with it), it avoids lots of casting and thus points of failures when something is not cast/managed correctly. The vendor headers also represent this kind of addresses using integers, which simplifies their usage.